zebra-rpc/src/methods.rs

```diff
@@ -1073,27 +1084,47 @@ where
             #[cfg(feature = "getblocktemplate-rpcs")]
             mempool::Response::FullTransactions {
                 mut transactions,
-                transaction_dependencies: _,
+                transaction_dependencies,
                 last_seen_tip_hash: _,
             } => {
-                // Sort transactions in descending order by fee/size, using hash in serialized byte order as a tie-breaker
-                transactions.sort_by_cached_key(|tx| {
-                    // zcashd uses modified fee here but Zebra doesn't currently
-                    // support prioritizing transactions
-                    std::cmp::Reverse((
-                        i64::from(tx.miner_fee) as u128 * MAX_BLOCK_BYTES as u128
-                            / tx.transaction.size as u128,
-                        // transaction hashes are compared in their serialized byte-order.
-                        tx.transaction.id.mined_id(),
-                    ))
-                });
-
-                let tx_ids: Vec<String> = transactions
-                    .iter()
-                    .map(|unmined_tx| unmined_tx.transaction.id.mined_id().encode_hex())
-                    .collect();
+                // verbose=true returns a map, where order does not matter,
+                // so we only need to sort when verbose=false.
+                if !verbose {
```

Comment:
Nitpick/Optional: It would be nice to move this condition to a match guard. The `if verbose` below could be an `else` on this if statement, and the `else` block below could be moved up to this block here.

Reply:
I'm not sold on the match guard, I think it would introduce an additional indentation level for little gain.

But I did follow your suggestions and moved things around to be clearer in 71c15241a

What do you think?